src/content/1.7/development/components/grid/tutorials/work-with-search-form/custom-form-factory.md

```diff
@@ -0,0 +1,146 @@
+     * @param UrlGeneratorInterface $urlGenerator
+     * @param RequestStack $requestStack
+     */
+    public function __construct(
+        GridFilterFormFactoryInterface $formFactory,
+        UrlGeneratorInterface $urlGenerator,
+        RequestStack $requestStack
+    ) {
+        $this->formFactory = $formFactory;
+        $this->urlGenerator = $urlGenerator;
+        $this->requestStack = $requestStack;
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function create(GridDefinitionInterface $definition)
+    {
+        // Use the default factory to build the form like usual
+        $categoryFilterForm = $this->formFactory->create($definition);
+
+        // Create a new empty form that will be used as an empty shell
+        $newCategoryFormBuilder = $categoryFilterForm->getConfig()->getFormFactory()->createNamedBuilder(
+            $definition->getId(),
+            FormType::class
+        );
+
+        // Adds all the form types in the new form
+        /** @var FormInterface $categoryFormItem */
+        foreach ($categoryFilterForm as $categoryFormItem) {
+            $newCategoryFormBuilder->add(
+                $categoryFormItem->getName(),
+                get_class($categoryFormItem->getConfig()->getType()->getInnerType()),
+                $categoryFormItem->getConfig()->getOptions()
+            );
+        }
+
+        $queryParams = [];
+
+        if (null !== ($request = $this->requestStack->getCurrentRequest())
+            && $request->query->has('id_category')
+        ) {
+            $queryParams['id_category'] = $request->query->get('id_category');
+        }
+
+        // Set the specific action in the new form
+        $newCategoryFormBuilder->setAction(
+            $this->urlGenerator->generate('admin_categories_search', $queryParams)
+        );
+
+        return $newCategoryFormBuilder->getForm();
+    }
+}
+```
+
+### Service definitions
+
+```yaml
+    # Define form factory decorator
+    prestashop.core.grid.filter.category_form_factory:
+        class: 'PrestaShop\PrestaShop\Core\Grid\Filter\CategoryFilterFormFactory'
+        arguments:
+            - '@prestashop.core.grid.filter.form_factory'
+            - '@router'
+            - '@request_stack'
+
+    # Use the decorator in the grid factory instead of the default one
+    prestashop.core.grid.factory.category:
+        class: 'PrestaShop\PrestaShop\Core\Grid\GridFactory'
+        arguments:
+            - '@prestashop.core.grid.definition.factory.category'
+            - '@prestashop.core.grid.data.factory.category_decorator'
+            - '@prestashop.core.grid.filter.category_form_factory'
+            - '@prestashop.core.hook.dispatcher'
+```
+
+## Simply change the action
+{{< minver v="1.7.7" title="true" >}}
+
+If you just need to change the form action PrestaShop provides a `FilterFormFactoryFormActionDecorator` decorator that simply sets the action you need to use. All you need to do is define the services properly:
```

Comment:
```diff
- simply
+ 
```

Always try to avoid "simply", "easily", "obviously"... I used to do the same mistake, and this make the reader feel (s)he's bad/stupid for free. Let's assume all topics are difficult for people who don't master it, yet :)